### PR TITLE
output: convert to posixpath if repo.fs is gitfs

### PR DESCRIPTION
**Pull request by skshetry**
Otherwise, metrics/params from revisions can't be read if the paths in dvc.yaml/.dvc files are using ntpath.

Something like following would not work if read from revisions on Windows, as `gitfs` uses slash. And `metrics/params diff` would fail to work. 

```yaml
stages:
  stage1:
    cmd:
    - mkdir -p dir2/sub
    - echo 'foo: 42' > dir2/sub/metrics.yaml
    wdir: dir
    metrics:
    - dir2\sub\metrics.yaml
```

Note that the Windows path don't work on Linux/macOS, but DVC writes `dvc.yaml`/`.dvc` files using those native paths. See https://github.com/iterative/dvc/issues/2059.

Also note that we have always translated `wdir` into a posixpath.

### CODE REVIEW

File: dvc/output.py [hunk 1]
```diff
@@ -474,7 +480,13 @@ def _parse_path(self, fs, fs_path):
             #
             # FIXME: if we have Windows path containing / or posix one with \
             # then we have #2059 bug and can't really handle that.
-            fs_path = fs.path.join(self.stage.wdir, fs_path)
+            wdir = self.stage.wdir
+            if isinstance(fs, GitFileSystem):
+                fs_path = as_posix(fs_path)
+                # dvc always generates wdir as posixpath but converting here
+                # just to be safe if it was written manually
+                wdir = as_posix(wdir)
```
Comment (skshetry) [inline dvc/output.py:488]: I can make this a bit weaker by only calling `as_posix` on Windows. But we are using this kind of translations liberally everywhere where we use dvcfs.

**Comment by efiop** [comment]
> Otherwise, metrics/params from revisions can't be read if the paths in dvc.yaml/.dvc files are using ntpath.

dvc will convert to posixpath when writing itself, right? So this situation only happens when someone is modifying stuff by hand?

**Comment by skshetry** [comment]
> dvc will convert to posixpath when writing itself, right? So this situation only happens when someone is modifying stuff by hand?

Only `wdir` is converted to posixpath, other paths are generated as native paths. See https://github.com/iterative/dvc/issues/7056.

File: dvc/output.py (hunk 1 above)
Comment (skshetry) [reply dvc/output.py:488]: We need to convert to `as_posix` at one point anyway.

**Comment by efiop** [comment]
For the record: discussed with the team today and decided to fix dvcfile generation to always write posixpaths (as it should've been and as we've always thought it has been) and that should be enough for now. People writing their dvcfiles by hand should use posixpaths, which we should probably document.

**Pull request closed without merging**